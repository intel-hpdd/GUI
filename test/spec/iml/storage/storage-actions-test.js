// @flow

import {
  addStorageResourceClasses,
  addStorageResources,
  setStorageSelectIndex,
  setStorageTableLoading,
  setStorageConfig
} from "../../../../source/iml/storage/storage-actions.js";

describe("storage actions", () => {
  it("should have a addStorageResourceClasses action", () => {
    expect(addStorageResourceClasses([])).toMatchSnapshot();
  });

  it("should have a addStorageResources action", () => {
    expect(
      addStorageResources({
        objects: []
      })
    ).toMatchSnapshot();
  });

  it("should have a setStorageResourceClassIndex action", () => {
    expect(setStorageSelectIndex(2)).toMatchSnapshot();
  });

  it("should have a setStorageTableLoading action", () => {
    expect(setStorageTableLoading(true)).toMatchSnapshot();
  });

  it("should have a setStorageConfig action", () => {
    expect(setStorageConfig({})).toMatchSnapshot();
  });
});
