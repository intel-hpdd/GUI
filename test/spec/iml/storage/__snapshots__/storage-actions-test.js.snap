// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`storage actions should have a addStorageResourceClasses action 1`] = `
Object {
  "payload": Array [],
  "type": "ADD_STORAGE_RESOURCE_CLASSES",
}
`;

exports[`storage actions should have a addStorageResources action 1`] = `
Object {
  "payload": Object {
    "objects": Array [],
  },
  "type": "ADD_STORAGE_RESOURCES",
}
`;

exports[`storage actions should have a setStorageConfig action 1`] = `
Object {
  "payload": Object {},
  "type": "SET_STORAGE_CONFIG",
}
`;

exports[`storage actions should have a setStorageResourceClassIndex action 1`] = `
Object {
  "payload": 2,
  "type": "SET_STORAGE_SELECT_INDEX",
}
`;

exports[`storage actions should have a setStorageTableLoading action 1`] = `
Object {
  "payload": true,
  "type": "SET_STORAGE_TABLE_LOADING",
}
`;
